### PR TITLE
refactor the navigation

### PR DESCRIPTION
**Pull request by aaronmarkham**
## Summary
* Simplifies the navigation; it is currently really cluttered.
* Organizes the files into subdirectories
* Fixes a couple link refs (Sphinx wasn't building after I reorganized... not sure why it worked before...)
* I may have fixed a typo here and there, but the content is otherwise untouched.
* I kept k8s and airflow in their own directories, but could have nested them in an /orchestration folder. I did it this way to demo a bit of flexibility. We can keep things relatively flat with only a few subfolders or nest as much as we think is better organized.

Happy to discuss what to pull out as a main node vs nesting...
I also didn't touch the Using... guide overview page... that needs some rework, but like I said, I didn't change the content for this PR, so it's somewhat easier to review in that regard.

## Notes
I think one thing that will help people contribute is just to add a new folder for a new topic, and use the automatic navigation provide by implementing the `toctree with `glob` like this on an index.rst page in any arbitrary folder.

```
############
Utility APIs
############

.. toctree::
   :maxdepth: 1
   :glob:

   *

```

## Preview
http://54.225.7.147/index.html

I think this is a lot better than what seems to be an unsorted list of random articles:

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: doc/algorithms/index.rst [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    factorization_machines
+    ipinsights
+    kmeans
+    knn
+    lda
+    linear_learner
+    ntm
+    object2vec
+    pca
+    randomcutforest
```
Comment (laurenyu) [inline doc/algorithms/index.rst:20]: doesn't have to be in this PR, but is there a way to automate getting the list of algorithms (maybe using the glob trick)?

File: doc/index.rst [hunk 2]
```diff
@@ -20,204 +21,59 @@ Overview
...
+Orchestration
+*************
+Orchestrate your SageMaker training and inference jobs with Kubernetes and Airflow.
 
 .. toctree::
     :maxdepth: 1
 
-    using_workflow
+    kubernetes/index
+    airflow/index
```
Comment (laurenyu) [inline doc/index.rst:64]: doesn't have to be in this PR, but to respond to the bit in the PR description about organizing the orchestration stuff, I think it probably would make sense to put these in an "orchestration" folder together

File: doc/index.rst [hunk 3]
```diff
@@ -20,204 +21,59 @@ Overview
...
 
 .. toctree::
     :maxdepth: 2
 
-    sagemaker.workflow.airflow
+    amazon_sagemaker_debugger
+
 
 *********************************
 Amazon SageMaker Model Monitoring
```
Comment (laurenyu) [inline doc/index.rst:79]: does it make sense to keep these last couple sections (model monitoring and processing) even though they've also been moved to the training/inference folders?

File: doc/algorithms/index.rst (hunk 1 above)
Comment (aaronmarkham) [reply doc/algorithms/index.rst:20]: yes, we can totally do that ... it'll be alphabetical, so if we want any organization, then we'd want to rename the files to fit that org.

**Pull request merged**